src/main/common/maths.c

```diff
@@ -30,53 +30,94 @@
-#define sinPolyCoef7 -1.849218155e-4f
-#define sinPolyCoef9  0
-#else
-#define sinPolyCoef3 -1.666665710e-1f                                          // Double: -1.666665709650470145824129400050267289858e-1
-#define sinPolyCoef5  8.333017292e-3f                                          // Double:  8.333017291562218127986291618761571373087e-3
-#define sinPolyCoef7 -1.980661520e-4f                                          // Double: -1.980661520135080504411629636078917643846e-4
-#define sinPolyCoef9  2.600054768e-6f                                          // Double:  2.600054767890361277123254766503271638682e-6
-#endif
+#if defined(FAST_MATH)
+static inline float sin_poly5_r(float r)
+{
+    // Pre-scaled for u = r*(π/2)
+    const float c0 =  0x1.921f1cp0f; // 1.5707871913909912109375
+    const float c1 = -0x1.4a974p-1f; // -0.6456851959228515625
+    const float c2 =  0x1.3db294p-4f; // 7.756288349628448486328125e-2
+    float s = r * r;
+    return r * ((c2 * s + c1) * s + c0);
+}
 
-float sin_approx(float x)
+static inline float cos_poly6_r(float r)
+{
+    const float d1 = -0x1.3bd39cp0f; // -1.2336976528167724609375
+    const float d2 =  0x1.03bp-2f; // 0.25360107421875
+    const float d3 = -0x1.4e5eecp-6f; // -2.04083733260631561279296875e-2
+    float s = r * r;
+    return ((d3 * s + d2) * s + d1) * s + 1.0f;
+}
+
+// ---- Quadrant mapping helpers ----
+// r ∈ [-0.5, 0.5], q is quadrant index (…,-1,0,1,2,3,4,…).
+static inline float sinf_quadrant_r(float r, int q)
 {
-    // Wrap angle to 2π with range [-π π]
-    x = fmodf(x, 2.0f * M_PIf);
-    // TO DO: these 'while' functions are not put into ITCM ram with current compiler flags
-    // A simpler 'if' function works, but gets put into ITCM ram, the extra 4% overflows F7xx ITCM
-    // The while function is retained only to avoid ITCM overflow for now
-    // ideally we should use the most efficient method, since sin_approx is used a LOT
-    // if (x <= -M_PIf) x += 2.0f * M_PIf;
-    // if (x > M_PIf) x -= 2.0f * M_PIf;
-    while (x >  M_PIf) x -= (2.0f * M_PIf);   // always wrap input angle to -PI..PI
-    while (x < -M_PIf) x += (2.0f * M_PIf);
-
-    // Use axis symmetry around x = ±π/2 for polynomial outside of range [-π/2 π/2]
-    if (x > M_PIf / 2) {
-        x = M_PIf - x; // Reflect
-    } else if (x < -M_PIf / 2) {
-        x = -M_PIf - x; // Reflect
+    q &= 3;
+    if (q & 1) { // odd: use cos, sign handled below
+        float v = cos_poly6_r(r);
+        return (q & 2) ? -v : v;
+    } else {     // even: use sin
+        float v = sin_poly5_r(r);
+        return (q & 2) ? -v : v;
     }
+}
 
-    float x2 = x * x;
-    return x + x * x2 * (sinPolyCoef3 + x2 * (sinPolyCoef5 + x2 * (sinPolyCoef7 + x2 * sinPolyCoef9)));
+static inline float cosf_quadrant_r(float r, int q)
+{
+    q &= 3;
+    if (q & 1) { // odd: -sin, sign handled below
+        float v = -sin_poly5_r(r);
+        return (q & 2) ? -v : v;   // q=1 -> -sin, q=3 -> +sin
+    } else {     // even: cos
```

Comment:
<img width="400" height="390" alt="cos" src="https://github.com/user-attachments/assets/712c1ca4-5fba-44d6-b95e-e20a24bff9e5" />

`v = -sin`, quadrant 3 is inverted again. 
`cos(x) = -sin(x-90)` -> `v = -sin` does make some sense.

Reply:
Ah yes, I missed the `-` in the `v = -sin`.
I'd prefer:
```c
float v = sin_poly5_r(r);
return (q & 2) ? v : -v;   // q=3 -> sin, q=1 -> -sin
 ```